src/index.cpp

```diff
@@ -2100,7 +2097,7 @@ std::pair<uint32_t, uint32_t> Index<T, TagT, LabelT>::search_with_filters(const
                 TagT tag;
                 if (_location_to_tag.try_get(best_L_nodes[i].id, tag))
                 {
-                    indices[pos] = (IdType)tag;
```

Comment:
same here, i think the existing search_with_filters code itself has got index and tags confused. so please let us discuss with Gopal to finalize this.

Reply:
reverted, but still need to keep:
indices[pos] = best_L_nodes[i].id;

As best_L_nodes[i].id is uint32, we can't cast to tag_uint128